crates/proto/src/udp/udp_stream.rs

```diff
@@ -232,12 +230,23 @@ impl UdpSocket for tokio::net::UdpSocket {
         tokio::net::UdpSocket::bind(addr).await
     }
 
-    async fn recv_from(&mut self, buf: &mut [u8]) -> io::Result<(usize, SocketAddr)> {
-        self.recv_from(buf).await
+    // TODO: add poll_recv_from and poll_send_to to be more efficient in allocations...
+    async fn recv_from(&self, buf: &mut [u8]) -> io::Result<(usize, SocketAddr)> {
+        futures_util::future::poll_fn(|cx| {
```

Comment:
So why aren't we using `UdpSocket::recv_from()` here? Does it have something to do with the bug you were hunting for?

Reply:
Yeah, basically. https://github.com/tokio-rs/tokio/issues/3084#issuecomment-722344699

Reply:
In fact, we really want to make this method not `async fn` based on its usage, imo.

Reply:
See https://github.com/bluejekyll/trust-dns/pull/1262/commits/8a1bb63c6b811ebf0c35e94a950f309261763979 which adds the poll variants.